pkg/ring/kv/memberlist/tcp_transport.go

```diff
@@ -476,8 +511,8 @@ func (t *TCPTransport) PacketCh() <-chan *memberlist.Packet {
 // two-way communication with a peer.
 func (t *TCPTransport) DialTimeout(addr string, timeout time.Duration) (net.Conn, error) {
 	t.outgoingStreams.Inc()
+	c, err := t.getConnection(addr, t.cfg.PacketDialTimeout)
```

Comment:
We probably should use the local `timeout`:

```suggestion
	c, err := t.getConnection(addr, timeout)
```

Reply:
👍